tests/prod_env_tests/test_post_algorithms.py

```diff
@@ -91,3 +91,38 @@ def test_post_algorithm_error(algorithm_name, request_dict, expected_response):
         response.status_code == exp_response_status
     ), f"Response message: {response.text}"
     assert re.search(exp_response_message, response.text)
+
+
+def test_post_algorithm_with_request_id():
+    algorithm_name = "logistic_regression"
+    request_dict = {
+        "inputdata": {
+            "y": ["alzheimerbroadcategory"],
+            "x": ["leftsogsuperioroccipitalgyrus", "rightmfcmedialfrontalcortex"],
+            "data_model": "dementia:0.1",
+            "datasets": [
+                "desd-synthdata8",
+                "desd-synthdata0",
+                "ppmi0",
+                "edsd8",
+                "desd-synthdata5",
+                "ppmi7",
+                "desd-synthdata1",
+                "edsd4",
+                "desd-synthdata4",
+                "edsd2",
+                "ppmi6",
+                "ppmi9",
+            ],
+            "filters": None,
+        },
+        "parameters": {"positive_class": "AD"},
+        "request_id": "89aace55-60e8-4b29-958b-84cca8785120",
+    }
+    algorithm_url = algorithms_url + "/" + algorithm_name
+    headers = {"Content-type": "application/json", "Accept": "text/plain"}
+    request_json = json.dumps(request_dict)
+    response = requests.post(algorithm_url, data=request_json, headers=headers)
+    assert response.status_code == 200, pytest.fail(
+        f"Algorithm did not succeed with {response.status_code=} and {response.text=}"
+    )
```

Comment:
Do you really need an e2e test, and with the slowest algorithm, to test if the one-line function `is_alnumhyphen` works as expected? Why not a few unit tests, some of which will also test for invalid strings? Like it's done with all other validators.

Reply:
Because we could miss something that we haven't consider. This test is testing that MIP-Engine as a whole is properly returning a result when a request id is provided.

If we had this test previously we would have seen the problem and we can't always know when this use case will behave improperly to always add unit tests.

Reply:
I will change the algorithm to a faster one. pca?